cocos2d/particle/CCSGParticleSystem.js

```diff
@@ -1761,8 +1762,12 @@ _ccsg.ParticleSystem = _ccsg.Node.extend({
                     var newPos = tpa;
                     if (this.positionType === _ccsg.ParticleSystem.Type.FREE || this.positionType === _ccsg.ParticleSystem.Type.RELATIVE) {
                         var diff = tpb;
-                        cc.pIn(diff, currentPosition);
-                        cc.pSubIn(diff, selParticle.startPos);
+                        // current Position convert To Node Space
+                        var localCurrentPosition = cc.pointApplyAffineTransform(currentPosition, worldToNodeTransform);
```

Comment:
在 CCAffineTransform.js 中修改：

```
cc._pointApplyAffineTransform = function (x, y, t) {   //it will remove.
    return cc.pointApplyAffineTransform(x, y, t);
};
```

为

```
cc._pointApplyAffineTransformIn = function (point, transOrY, transOrOut, out) {
    var x, y, t;
    if (out === undefined) {
        t = transOrY;
        x = point.x;
        y = point.y;
        out = transOrOut;
    } else {
        x = point;
        y = transOrY;
        t = transOrOut;
    }
    out.x = t.a * x + t.c * y + t.tx;
    out.y = t.b * x + t.d * y + t.ty;
};
```

用这个不创建对象的函数来计算